client/src/partials/accounts/accounts.js

```diff
@@ -2,192 +2,325 @@ angular.module('bhima.controllers')
 .controller('AccountsController', AccountsController);
 
 AccountsController.$inject = [
-  '$scope', 'validate', 'appstate', 'connect', '$translate',
-  'liberror', 'messenger'
+  'validate', 'appstate', 'connect',
+  '$translate', 'liberror', 'messenger',
+  'Account', 'AccountType', 'costCenter', 
+  'profitCenter', 'reference', 'SessionService'
 ];
 
-function AccountsController($scope, validate, appstate, connect, $translate, liberror, messenger) {
-  var dependencies = {}, titleAccount = 3;
-  var grid, columns, options, dataview, sortColumn = 'account_number';
-  var session = $scope.session = { state: 'display' };
-
-  var accountError = liberror.namespace('ACCOUNT');
-
-  $scope.newAccount = {};
-  $scope.editAccount = {};
-
-  dependencies.account = {
-    required : true,
-    query : {
-      identifier : 'account_number',
-      tables : {
-        account : { columns : ['id', 'account_number', 'account_txt', 'account_type_id', 'cc_id', 'pc_id', 'is_asset', 'is_ohada', 'parent', 'locked', 'reference_id', 'is_brut_link', 'is_used_budget', 'classe', 'is_charge'] },
-        account_type : { columns : ['type::account_type'] }
-      },
-      join: [ 'account.account_type_id=account_type.id' ]
-    }
-  };
+function AccountsController(validate, appstate, connect, $translate, liberror, messenger, accountService, accountTypeService, costCenterService, profitCenterService, referenceService, sessionService) {
+  var vm = this;
 
-  dependencies.references = {
-    query : {
-      identifier : 'id',
-      tables : {
-        reference : { columns : ['id', 'ref', 'text', 'position', 'reference_group_id', 'section_resultat_id', 'is_report'] }
-      }
-    }
+  vm.session = {
+    filterAll : false,
+    state : null,
+    filter : null
   };
 
-  dependencies.costCenter = {
-    query : {
-      tables : {
-        cost_center : { columns : ['id', 'text'] },
-        project     : { columns : ['abbr']}
-      },
-      join : ['cost_center.project_id=project.id']
-    }
-  };
+  vm.newAccount = {};
+  vm.editable_account = {};
```

Comment:
You should use camelCase for all variables.  Mixing camelCase and underscore_case is confusing.